ngraph/core/include/ngraph/op/util/sub_graph_base.hpp

```diff
@@ -5,187 +5,24 @@
-                    ///
-                    /// \param      body_value_index  A body value that produces the output
-                    /// \param      output_index      The SubGraphOp output index
-                    ///
-                    OutputDescription(uint64_t body_value_index, uint64_t output_index);
-                    OutputDescription() = default;
-
-                public:
-                    using type_info_t = DiscreteTypeInfo;
-                    virtual ~OutputDescription() = default;
-                    virtual std::shared_ptr<OutputDescription> copy() const = 0;
-                    virtual const type_info_t& get_type_info() const = 0;
-
-                    uint64_t m_body_value_index{0};
-                    uint64_t m_output_index{0};
-                };
-
-                /// \brief Produces an output by concatenating an output from each iteration
-                class NGRAPH_API ConcatOutputDescription : public OutputDescription
-                {
-                public:
-                    static constexpr type_info_t type_info{"ConcatOutputDescription", 0};
-                    const type_info_t& get_type_info() const override { return type_info; }
-                    ///
-                    /// \brief      Constructs a new instance.
-                    ///
-                    /// \param      body_value_index  A body value that produces the output
-                    /// \param      output_index      The SubGraphOp output index
-                    /// \param      start             First index for slices
-                    /// \param      stride            Step amount for slices
-                    /// \param      part_size         Width of slices
-                    /// \param      end               Last index for slices
-                    /// \param      axis              Axis being sliced
-                    ///
-                    ConcatOutputDescription(uint64_t body_value_index,
-                                            uint64_t output_index,
-                                            int64_t start,
-                                            int64_t stride,
-                                            int64_t part_size,
-                                            int64_t end,
-                                            int64_t axis);
-                    ConcatOutputDescription() = default;
-
-                    std::shared_ptr<OutputDescription> copy() const override;
-                    int64_t m_start{0};
-                    int64_t m_stride{0};
-                    int64_t m_part_size{0};
-                    int64_t m_end{0};
-                    int64_t m_axis{0};
-                };
 
                 /// \brief Produces an output from a specific iteration
-                class NGRAPH_API BodyOutputDescription : public OutputDescription
+                class NGRAPH_API BodyOutputDescription : public MultiSubGraphOp::OutputDescription
```

Comment:
can we move it to MultiSubgraph and add -1 as default value for iteration?
` BodyOutputDescription(uint64_t body_value_index,
                                          uint64_t output_index,
                                          int64_t iteration = -1);`